File: result/translate/starchat/java2cpp/c++/MAXIMUM_LENGTH_SUBSEQUENCE_DIFFERENCE_ADJACENT_ELEMENTS_EITHER_0_1.cpp
int f_gold ( int arr [ ], int n ) {
        vector<int> mls(n,1);
        for (int i = 1; i < n; i++)
            for (int j = 0; j < i; j++)
                if (abs(arr[i] - arr[j]) <= 1 && mls[i] < mls[j] + 1)
                    mls[i] = mls[j] + 1;
        int max = *max_element(mls.begin(), mls.end());
        return max;
    }

File: result/translate/starchat/java2python/python/MAXIMUM_LENGTH_SUBSEQUENCE_DIFFERENCE_ADJACENT_ELEMENTS_EITHER_0_1.py
def f_gold(arr, n):
        mls = [1] * n
        for i in range(n):
            for j in range(i):
                if abs(arr[i] - arr[j]) <= 1 and mls[i] < mls[j] + 1:
                    mls[i] = mls[j] + 1
        max_len = max(mls)
        return max_len